src/Uno.UI/UI/Xaml/Media/Animation/Animators/RenderingLoopAnimator.wasm.cs

```diff
@@ -1,130 +1,112 @@
+	protected override void EnableFrameReporting()
+	{
+		if (this.Log().IsEnabled(LogLevel.Trace))
 		{
-			if (Handle.IsAlive)
-			{
-				NativeMethods.EnableFrameReporting(Handle.JSHandle);
-			}
-			else if (this.Log().IsEnabled(LogLevel.Debug))
-			{
-				this.Log().Debug("Cannot EnableFrameReporting as Handle is no longer alive.");
-			}
+			this.Log().Trace("EnableFrameReporting");
 		}
 
-		protected override void DisableFrameReporting()
+		if (_isEnabled)
 		{
-			if (Handle.IsAlive)
-			{
-				NativeMethods.DisableFrameReporting(Handle.JSHandle);
-			}
-			else if (this.Log().IsEnabled(LogLevel.Debug))
-			{
-				this.Log().Debug("Cannot DisableFrameReporting as Handle is no longer alive.");
-			}
+			return;
 		}
 
-		protected override void SetStartFrameDelay(long delayMs)
-		{
-			if (Handle.IsAlive)
-			{
-				NativeMethods.SetStartFrameDelay(Handle.JSHandle, delayMs);
-			}
-			else if (this.Log().IsEnabled(LogLevel.Debug))
-			{
-				this.Log().Debug("Cannot SetStartFrameDelay as Handle is no longer alive.");
-			}
-		}
-
-		protected override void SetAnimationFramesInterval()
-		{
-			if (Handle.IsAlive)
-			{
-				NativeMethods.SetAnimationFramesInterval(Handle.JSHandle);
-			}
-			else if (this.Log().IsEnabled(LogLevel.Debug))
-			{
-				this.Log().Debug("Cannot SetAnimationFramesInterval as Handle is no longer alive.");
-			}
-		}
+		_isEnabled = true;
 
-		private void OnFrame() => OnFrame(null, null);
+		_frameEvent = RenderingLoopAnimator.RegisterFrameEvent(OnFrame);
+	}
 
-		/// <inheritdoc />
-		public override void Dispose()
+	protected override void DisableFrameReporting()
+	{
+		if (this.Log().IsEnabled(LogLevel.Trace))
 		{
-			// WARNING: If the Dispose is invoked by the GC, it has most probably already disposed the Handle,
-			//			which means that we have already lost ability to dispose/stop the native object!
+			this.Log().Trace("DisableFrameReporting");
+		}
 
-			base.Dispose();
-			Handle.Dispose();
 
-			GC.SuppressFinalize(this);
-		}
+		_isEnabled = false;
+		UnscheduleFrame();
+	}
 
-		~RenderingLoopAnimator()
+	protected override void SetStartFrameDelay(long delayMs)
+	{
+		if (this.Log().IsEnabled(LogLevel.Trace))
 		{
-			Dispose();
+			this.Log().Trace($"SetStartFrameDelay: {delayMs}");
 		}
 
-		private class Metadata : IJSObjectMetadata
+		UnscheduleFrame();
+
+		if (_isEnabled)
 		{
-			public static Metadata Instance { get; } = new Metadata();
+			_delayRequest = new DispatcherTimer() { Interval = TimeSpan.FromMilliseconds(delayMs) };
+			_delayRequest.Tick += (s, e) =>
+			{
+				_delayRequest.Stop();
+				_delayRequest = null;
 
-			private Metadata() { }
+				OnFrame();
 
-			/// <inheritdoc />
-			public long CreateNativeInstance(IntPtr managedHandle)
-			{
-				var id = RenderingLoopAnimatorMetadataIdProvider.Next();
+				// Restore the render loop now that we have completed the delay
+				_frameEvent = RenderingLoopAnimator.RegisterFrameEvent(OnFrame);
```

Comment:
Not sure if it's possible, but if `EnableFrameReporting` has been invoked we might lost a subscription an leak:
```suggestion
				_frameEvent ??= RenderingLoopAnimator.RegisterFrameEvent(OnFrame);
```